From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xiaoyinl <xiaoyinl@users.noreply.github.com>
Date: Mon, 22 Dec 2025 07:31:00 +0800
Subject: [PATCH] 8815: Update Linux syscall numbers

Update x86_linux_syscall_numbers

Update x86 Linux syscall. The command to generate this file is: curl --no-progress-meter "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/plain/arch/x86/entry/syscalls/syscall_32.tbl?h=linux-rolling-stable" | sed --expression='/^#/d' | cut -f 1,3 | expand -t 1

Update x64_linux_syscall_numbers

New x64_linux_syscall_numbers is generated by: curl --no-progress-meter "https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/plain/arch/x86/entry/syscalls/syscall_64.tbl?h=linux-rolling-stable" | awk 'NF && !/^#/ && $2 != "x32" { print $1,$3 }'

Syscall numbers 512-547 are removed because they are x32 abi.
---
 .../Base/data/x64_linux_syscall_numbers       |  64 ++-
 .../Base/data/x86_linux_syscall_numbers       | 366 +++++++++++++++---
 2 files changed, 380 insertions(+), 50 deletions(-)

diff --git a/Ghidra/Features/Base/data/x64_linux_syscall_numbers b/Ghidra/Features/Base/data/x64_linux_syscall_numbers
index 2a38eee37d..4771b10a23 100644
--- a/Ghidra/Features/Base/data/x64_linux_syscall_numbers
+++ b/Ghidra/Features/Base/data/x64_linux_syscall_numbers
@@ -11,9 +11,10 @@
 9 mmap
 10 mprotect
 11 munmap
-12 sys_brk
+12 brk
 13 rt_sigaction
 14 rt_sigprocmask
+15 rt_sigreturn
 16 ioctl
 17 pread64
 18 pwrite64
@@ -203,12 +204,17 @@
 202 futex
 203 sched_setaffinity
 204 sched_getaffinity
+205 set_thread_area
 206 io_setup
 207 io_destroy
 208 io_getevents
 209 io_submit
 210 io_cancel
+211 get_thread_area
+212 lookup_dcookie
 213 epoll_create
+214 epoll_ctl_old
+215 epoll_wait_old
 216 remap_file_pages
 217 getdents64
 218 set_tid_address
@@ -229,6 +235,7 @@
 233 epoll_ctl
 234 tgkill
 235 utimes
+236 vserver
 237 mbind
 238 set_mempolicy
 239 get_mempolicy
@@ -308,15 +315,70 @@
 313 finit_module
 314 sched_setattr
 315 sched_getattr
+316 renameat2
 317 seccomp
 318 getrandom
 319 memfd_create
 320 kexec_file_load
 321 bpf
+322 execveat
 323 userfaultfd
 324 membarrier
 325 mlock2
 326 copy_file_range
+327 preadv2
+328 pwritev2
 329 pkey_mprotect
 330 pkey_alloc
 331 pkey_free
+332 statx
+333 io_pgetevents
+334 rseq
+335 uretprobe
+336 uprobe
+424 pidfd_send_signal
+425 io_uring_setup
+426 io_uring_enter
+427 io_uring_register
+428 open_tree
+429 move_mount
+430 fsopen
+431 fsconfig
+432 fsmount
+433 fspick
+434 pidfd_open
+435 clone3
+436 close_range
+437 openat2
+438 pidfd_getfd
+439 faccessat2
+440 process_madvise
+441 epoll_pwait2
+442 mount_setattr
+443 quotactl_fd
+444 landlock_create_ruleset
+445 landlock_add_rule
+446 landlock_restrict_self
+447 memfd_secret
+448 process_mrelease
+449 futex_waitv
+450 set_mempolicy_home_node
+451 cachestat
+452 fchmodat2
+453 map_shadow_stack
+454 futex_wake
+455 futex_wait
+456 futex_requeue
+457 statmount
+458 listmount
+459 lsm_get_self_attr
+460 lsm_set_self_attr
+461 lsm_list_modules
+462 mseal
+463 setxattrat
+464 getxattrat
+465 listxattrat
+466 removexattrat
+467 open_tree_attr
+468 file_getattr
+469 file_setattr
diff --git a/Ghidra/Features/Base/data/x86_linux_syscall_numbers b/Ghidra/Features/Base/data/x86_linux_syscall_numbers
index 7b83ea1da0..172e90be00 100644
--- a/Ghidra/Features/Base/data/x86_linux_syscall_numbers
+++ b/Ghidra/Features/Base/data/x86_linux_syscall_numbers
@@ -1,14 +1,14 @@
 #format = number(decimal) syscall_name
-00 setup 
-01 exit
-02 fork
-03 read
-04 write
-05 open
-06 close
-07 waitpid
-08 creat
-09 link
+0 restart_syscall
+1 exit
+2 fork
+3 read
+4 write
+5 open
+6 close
+7 waitpid
+8 creat
+9 link
 10 unlink
 11 execve
 12 chdir
@@ -34,7 +34,7 @@
 32 gtty
 33 access
 34 nice
-35 ftime 
+35 ftime
 36 sync
 37 kill
 38 rename
@@ -43,7 +43,7 @@
 41 dup
 42 pipe
 43 times
-44 prof 
+44 prof
 45 brk
 46 setgid
 47 getgid
@@ -51,13 +51,13 @@
 49 geteuid
 50 getegid
 51 acct
-52 umount2 
-53 lock 
+52 umount2
+53 lock
 54 ioctl
 55 fcntl
-56 mpx 
+56 mpx
 57 setpgid
-58 ulimit 
+58 ulimit
 59 oldolduname
 60 umask
 61 chroot
@@ -81,7 +81,7 @@
 79 settimeofday
 80 getgroups
 81 setgroups
-82 select 
+82 select
 83 symlink
 84 oldlstat
 85 readlink
@@ -89,7 +89,7 @@
 87 swapon
 88 reboot
 89 readdir
-90 mmap 
+90 mmap
 91 munmap
 92 truncate
 93 ftruncate
@@ -97,7 +97,7 @@
 95 fchown
 96 getpriority
 97 setpriority
-98 profil 
+98 profil
 99 statfs
 100 fstatfs
 101 ioperm
@@ -105,10 +105,10 @@
 103 syslog
 104 setitimer
 105 getitimer
-106 stat 
+106 stat
 107 lstat
 108 fstat
-109 olduname 
+109 olduname
 110 iopl
 111 vhangup
 112 idle
@@ -121,7 +121,7 @@
 119 sigreturn
 120 clone
 121 setdomainname
-122 uname 
+122 uname
 123 modify_ldt
 124 adjtimex
 125 mprotect
@@ -136,19 +136,19 @@
 134 bdflush
 135 sysfs
 136 personality
-137 afs_syscall 
+137 afs_syscall
 138 setfsuid
 139 setfsgid
-140 _llseek 
+140 _llseek
 141 getdents
-142 _newselect 
+142 _newselect
 143 flock
 144 msync
 145 readv
 146 writev
 147 getsid
 148 fdatasync
-149 _sysctl 
+149 _sysctl
 150 mlock
 151 munlock
 152 mlockall
@@ -163,30 +163,298 @@
 161 sched_rr_get_interval
 162 nanosleep
 163 mremap
-164 setresuid 
+164 setresuid
 165 getresuid
 166 vm86
-167 query_module 
-168 poll 
+167 query_module
+168 poll
 169 nfsservctl
-170 setresgid 
-171 getresgid 
-172 prctl 
-173 rt_sigreturn 
-174 rt_sigaction 
-175 rt_sigprocmask 
-176 rt_sigpending 
-177 rt_sigtimedwait 
-178 rt_sigqueueinfo 
-179 rt_sigsuspend 
-180 pread 
-181 pwrite 
-182 chown 
-183 getcwd 
-184 capget 
-185 capset 
+170 setresgid
+171 getresgid
+172 prctl
+173 rt_sigreturn
+174 rt_sigaction
+175 rt_sigprocmask
+176 rt_sigpending
+177 rt_sigtimedwait
+178 rt_sigqueueinfo
+179 rt_sigsuspend
+180 pread64
+181 pwrite64
+182 chown
+183 getcwd
+184 capget
+185 capset
 186 sigaltstack
-187 sendfile 
-188 getpmsg 
-189 putpmsg 
-190 vfork 
+187 sendfile
+188 getpmsg
+189 putpmsg
+190 vfork
+191 ugetrlimit
+192 mmap2
+193 truncate64
+194 ftruncate64
+195 stat64
+196 lstat64
+197 fstat64
+198 lchown32
+199 getuid32
+200 getgid32
+201 geteuid32
+202 getegid32
+203 setreuid32
+204 setregid32
+205 getgroups32
+206 setgroups32
+207 fchown32
+208 setresuid32
+209 getresuid32
+210 setresgid32
+211 getresgid32
+212 chown32
+213 setuid32
+214 setgid32
+215 setfsuid32
+216 setfsgid32
+217 pivot_root
+218 mincore
+219 madvise
+220 getdents64
+221 fcntl64
+224 gettid
+225 readahead
+226 setxattr
+227 lsetxattr
+228 fsetxattr
+229 getxattr
+230 lgetxattr
+231 fgetxattr
+232 listxattr
+233 llistxattr
+234 flistxattr
+235 removexattr
+236 lremovexattr
+237 fremovexattr
+238 tkill
+239 sendfile64
+240 futex
+241 sched_setaffinity
+242 sched_getaffinity
+243 set_thread_area
+244 get_thread_area
+245 io_setup
+246 io_destroy
+247 io_getevents
+248 io_submit
+249 io_cancel
+250 fadvise64
+252 exit_group
+253 lookup_dcookie
+254 epoll_create
+255 epoll_ctl
+256 epoll_wait
+257 remap_file_pages
+258 set_tid_address
+259 timer_create
+260 timer_settime
+261 timer_gettime
+262 timer_getoverrun
+263 timer_delete
+264 clock_settime
+265 clock_gettime
+266 clock_getres
+267 clock_nanosleep
+268 statfs64
+269 fstatfs64
+270 tgkill
+271 utimes
+272 fadvise64_64
+273 vserver
+274 mbind
+275 get_mempolicy
+276 set_mempolicy
+277 mq_open
+278 mq_unlink
+279 mq_timedsend
+280 mq_timedreceive
+281 mq_notify
+282 mq_getsetattr
+283 kexec_load
+284 waitid
+286 add_key
+287 request_key
+288 keyctl
+289 ioprio_set
+290 ioprio_get
+291 inotify_init
+292 inotify_add_watch
+293 inotify_rm_watch
+294 migrate_pages
+295 openat
+296 mkdirat
+297 mknodat
+298 fchownat
+299 futimesat
+300 fstatat64
+301 unlinkat
+302 renameat
+303 linkat
+304 symlinkat
+305 readlinkat
+306 fchmodat
+307 faccessat
+308 pselect6
+309 ppoll
+310 unshare
+311 set_robust_list
+312 get_robust_list
+313 splice
+314 sync_file_range
+315 tee
+316 vmsplice
+317 move_pages
+318 getcpu
+319 epoll_pwait
+320 utimensat
+321 signalfd
+322 timerfd_create
+323 eventfd
+324 fallocate
+325 timerfd_settime
+326 timerfd_gettime
+327 signalfd4
+328 eventfd2
+329 epoll_create1
+330 dup3
+331 pipe2
+332 inotify_init1
+333 preadv
+334 pwritev
+335 rt_tgsigqueueinfo
+336 perf_event_open
+337 recvmmsg
+338 fanotify_init
+339 fanotify_mark
+340 prlimit64
+341 name_to_handle_at
+342 open_by_handle_at
+343 clock_adjtime
+344 syncfs
+345 sendmmsg
+346 setns
+347 process_vm_readv
+348 process_vm_writev
+349 kcmp
+350 finit_module
+351 sched_setattr
+352 sched_getattr
+353 renameat2
+354 seccomp
+355 getrandom
+356 memfd_create
+357 bpf
+358 execveat
+359 socket
+360 socketpair
+361 bind
+362 connect
+363 listen
+364 accept4
+365 getsockopt
+366 setsockopt
+367 getsockname
+368 getpeername
+369 sendto
+370 sendmsg
+371 recvfrom
+372 recvmsg
+373 shutdown
+374 userfaultfd
+375 membarrier
+376 mlock2
+377 copy_file_range
+378 preadv2
+379 pwritev2
+380 pkey_mprotect
+381 pkey_alloc
+382 pkey_free
+383 statx
+384 arch_prctl
+385 io_pgetevents
+386 rseq
+393 semget
+394 semctl
+395 shmget
+396 shmctl
+397 shmat
+398 shmdt
+399 msgget
+400 msgsnd
+401 msgrcv
+402 msgctl
+403 clock_gettime64
+404 clock_settime64
+405 clock_adjtime64
+406 clock_getres_time64
+407 clock_nanosleep_time64
+408 timer_gettime64
+409 timer_settime64
+410 timerfd_gettime64
+411 timerfd_settime64
+412 utimensat_time64
+413 pselect6_time64
+414 ppoll_time64
+416 io_pgetevents_time64
+417 recvmmsg_time64
+418 mq_timedsend_time64
+419 mq_timedreceive_time64
+420 semtimedop_time64
+421 rt_sigtimedwait_time64
+422 futex_time64
+423 sched_rr_get_interval_time64
+424 pidfd_send_signal
+425 io_uring_setup
+426 io_uring_enter
+427 io_uring_register
+428 open_tree
+429 move_mount
+430 fsopen
+431 fsconfig
+432 fsmount
+433 fspick
+434 pidfd_open
+435 clone3
+436 close_range
+437 openat2
+438 pidfd_getfd
+439 faccessat2
+440 process_madvise
+441 epoll_pwait2
+442 mount_setattr
+443 quotactl_fd
+444 landlock_create_ruleset
+445 landlock_add_rule
+446 landlock_restrict_self
+447 memfd_secret
+448 process_mrelease
+449 futex_waitv
+450 set_mempolicy_home_node
+451 cachestat
+452 fchmodat2
+453 map_shadow_stack
+454 futex_wake
+455 futex_wait
+456 futex_requeue
+457 statmount
+458 listmount
+459 lsm_get_self_attr
+460 lsm_set_self_attr
+461 lsm_list_modules
+462 mseal
+463 setxattrat
+464 getxattrat
+465 listxattrat
+466 removexattrat
+467 open_tree_attr
+468 file_getattr
+469 file_setattr
-- 
2.45.1

